packages/bug-detectors/index.ts

```diff
@@ -0,0 +1,188 @@
+ * @example
+ * registerBugDetectors(["commandInjection"], (e, n) => console.log(e.message))
+ * .then(() => console.log("All bug detectors registered."));
+ */
+export async function registerBugDetectors(
+	bugDetectors: string[],
+	saveFirstBugDetectorException: (e: Error, trimErrorStackLines: number) => void
+): Promise<void> {
+	// Keep track of registered bug detectors to avoid registering the same bug detector multiple times.
+	const registeredBugDetectors: string[] = [];
+	for (let i = 0; i < bugDetectors.length; i++) {
+		if (registeredBugDetectors.includes(bugDetectors[i])) {
+			continue;
+		}
+		switch (bugDetectors[i]) {
+			case "commandInjectionSafe":
+				registeredBugDetectors.push(bugDetectors[i]);
+				await registerCommandInjectionBugDetector(
+					saveFirstBugDetectorException,
+					false
+				);
+				break;
+			case "commandInjection":
+				registeredBugDetectors.push(bugDetectors[i]);
+				await registerCommandInjectionBugDetector(
+					saveFirstBugDetectorException,
+					true
+				);
+				break;
+		}
+	}
+}
+
+/**
+ * Replaces a built-in function with a custom implementation while preserving
+ * the original function for potential use within the replacement function.
+ *
+ * @template F - The type of the replacement function. Must extend Function.
+ * @template K - The type of the original function being replaced. Must extend Function.
+ * @param moduleName - The name of the module containing the target function.
+ * @param targetFnName - The name of the target function to be replaced.
+ * @param replacementFn - The replacement function that will be called instead
+ *                        of the original function. The first argument passed
+ *                        to the replacement function will be the original function,
+ *                        followed by any arguments that were originally passed
+ *                        to the target function.
+ * @returns A promise that resolves to the original function that was replaced.
+ * @throws Will throw an error if the module cannot be imported.
+ *
+ * @example
+ * const originalExec = await hookBuiltInFunction(
+ *   "child_process",
+ *   "exec",
+ *   (originalFn: Function, cmd: string, options: object, callback: Function) => {
+ *     console.log("Custom implementation called with command:", cmd);
+ *     return originalFn(cmd, options, callback);
+ *   }
+ * );
+ */
+export async function hookBuiltInFunction<
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	F extends Function,
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	K extends Function
+>(moduleName: string, targetFnName: string, replacementFn: F): Promise<K> {
+	const { default: module } = await import(moduleName);
+	const originalFn = module[targetFnName];
+	delete module[targetFnName];
+	module[targetFnName] = (...args: unknown[]) =>
+		replacementFn(originalFn, ...args);
+	return originalFn;
+}
+
+// TODO: Internally Jazzer.js should use original unhooked builtin functions.
```

Comment:
This will be important for the path traversal detectors. In safe mode Jest probably won't be able to make/check any directory!